nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -92,6 +92,8 @@ def __init__(
         self.shuffle_shards = shuffle_shards
         self.shard_seed = shard_seed
         paths = expand_sharded_filepaths(path)
+        cache_datastore_manifests(paths)
```

Comment:
Is this necessary? Wondering since this is not in `LazyNeMoTarredIterator`, and that lhotse should be able to automatically fetch and cache the data? @pzelasko what do you think?

Reply:
Lhotse supports URLs but doesn't perform caching. If we tried caching tarred data we would run out of local space in many cases. For non-tarred data it is probably fine as these datasets would typically be rather small.

I am not familiar with NeMo's original `cache_datastore_manifests` function, does it gracefully handle datasets larger than available local disk space? @monica-sekoyan  @stevehuang52